### DISCUSSION TITLE
Failed to acquire the "db:create" lock.

### DISCUSSION THREAD

**Discussion by onedr0p** [shlinkio/discussions#87] [Help wanted] [answered]
Hi 👋

I've been running shlink for a few months now without any issue. I recently had an unclean shutdown of my kubernetes cluster and shlink was the only app that didn't come back online. Here's the errors from the container. Any help would be appreciated 👍

```
❯ k logs shlink-api-6555fc4dd6-9k2vt
Creating fresh database if needed...
[2022-10-04T19:08:43.942799-04:00] [NULL] Shlink.NOTICE - Failed to acquire the "db:create" lock.

In Lock.php line 126:
          Failed to acquire the "db:create" lock.

In SentinelReplication.php line 560:
         Expected master but got slave [redis-node-0.redis-headless.default.svc.cluster.local:6379]

db:create
```

Configuration:

```yaml
    env:
      DB_DRIVER: postgres
      DB_HOST: postgres-rw.default.svc.cluster.local
      DB_NAME: shlink
      DEFAULT_DOMAIN: &host "ln.${SECRET_PUBLIC_DOMAIN}"
      DISABLE_TRACKING_FROM: 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16
      ENABLE_PERIODIC_VISIT_LOCATE: "true"
      IS_HTTPS_ENABLED: "true"
      PORT: &port 80
      REDIS_PUB_SUB_ENABLED: "true"
      REDIS_SENTINEL_SERVICE: redis-master
      REDIS_SERVERS: >-
        redis-node-0.redis-headless.default.svc.cluster.local:26379,
        redis-node-1.redis-headless.default.svc.cluster.local:26379,
        redis-node-2.redis-headless.default.svc.cluster.local:26379
      TIMEZONE: "${TIMEZONE}"
```

**Comment by acelaya** [comment]
Hmmm. To me it looks like a bug in the redis integration library. There are some similar reports https://github.com/predis/predis/issues?q=is%3Aissue+is%3Aopen+Expected+master+but+got+slave

I will have to investigate a bit.

**Reply by onedr0p** [reply]
It might have actually been on my end, I changed:

```yaml
      REDIS_SERVERS: >-
        redis-node-0.redis-headless.default.svc.cluster.local:26379,
        redis-node-1.redis-headless.default.svc.cluster.local:26379,
        redis-node-2.redis-headless.default.svc.cluster.local:26379
```

to:

```yaml
      REDIS_SERVERS: redis-node-0.redis-headless.default.svc.cluster.local:26379,redis-node-1.redis-headless.default.svc.cluster.local:26379,redis-node-2.redis-headless.default.svc.cluster.local:26379
```

My guess is the code doesn't like being comma+space delimited, as the first block would render out to be:

```yaml
      REDIS_SERVERS: redis-node-0.redis-headless.default.svc.cluster.local:26379, redis-node-1.redis-headless.default.svc.cluster.local:26379, redis-node-2.redis-headless.default.svc.cluster.local:26379
```

Notice the space after the comma. I guess it's fine to sacrifice prettiness for functionality sometimes :D

I am still curious as to why it worked this way for so long haha

**Reply by acelaya** [reply]
Interesting.

Maybe it turned out the first node was a master until your cluster restarted, making it work by pure luck.

I will make sure spaces are trimmed after splitting by comma. That should make your first approach work as well.

**Reply by onedr0p** [reply]
That makes sense and thanks for trimming the whitespace, it will make the config much easier to look at :)

**Reply by acelaya** [reply]
This should do the trick https://github.com/shlinkio/shlink-common/pull/109

It will be included in next Shlink release.

**Comment by acelaya** [comment, marked as answer]
> *From @onedr0p on https://github.com/shlinkio/shlink/discussions/1561#discussioncomment-3815684*
> 
> It might have actually been on my end, I changed:
> 
> ```yaml
>       REDIS_SERVERS: >-
>         redis-node-0.redis-headless.default.svc.cluster.local:26379,
>        redis-node-1.redis-headless.default.svc.cluster.local:26379,
>         redis-node-2.redis-headless.default.svc.cluster.local:26379
> ```
> 
> to:
> 
> ```yaml
>       REDIS_SERVERS: redis-node-0.redis-headless.default.svc.cluster.local:26379,redis-node-1.redis-headless.default.svc.cluster.local:26379,redis-node-2.redis-headless.default.svc.cluster.local:26379
> ```
